shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java

```diff
@@ -111,9 +120,22 @@ public InputConnection createInputConnection(FlutterView view, EditorInfo outAtt
             mConfiguration.getString("inputType"),
             mConfiguration.optBoolean("obscureText"),
             mConfiguration.optBoolean("autocorrect", true));
-        if (!mConfiguration.isNull("actionLabel"))
-          outAttrs.actionLabel = mConfiguration.getString("actionLabel");
-        outAttrs.imeOptions = EditorInfo.IME_ACTION_DONE | EditorInfo.IME_FLAG_NO_FULLSCREEN;
+        outAttrs.imeOptions = EditorInfo.IME_FLAG_NO_FULLSCREEN;
+        int enterAction;
+        if (mConfiguration.isNull("inputAction")) {
+            // If an explicit input action isn't set, then default to none for multi-line fields
+            // and done for single line fields.
+            enterAction =
+                (InputType.TYPE_TEXT_FLAG_MULTI_LINE & outAttrs.inputType) != 0 ?
+                    EditorInfo.IME_ACTION_NONE : EditorInfo.IME_ACTION_DONE;
```

Comment:
Personally I'd go for:
```java
enterAction = (InputType.TYPE_TEXT_FLAG_MULTI_LINE & outAttrs.inputType) != 0
    ? EditorInfo.IME_ACTION_NONE
    : EditorInfo.IME_ACTION_DONE;
```
but this is a great example of a comment that would have been prevented by us adopting an auto-formatter ;-)

Reply:
Done.  Yep.  I'm telling you, it eliminates 1/3 of the comments on PRs.